### PR TITLE
Add support and instructions for function calling models

### PR DESCRIPTION
**Pull request by gvspraveen**

### CODE REVIEW

File: templates/endpoints/llm-serve.yaml [hunk 1]
```diff
@@ -1,10 +1,11 @@
...
-  runtime_env:
-    env_vars:
-      HUGGING_FACE_HUB_TOKEN: "UPDATE HF TOKEN HERE" # needed for Llama 2 models
+  - name: llm-endpoint
+    route_prefix: /
+    import_path: aviary_private_endpoints.backend.server.run:router_application
+    args:
+      models: []
+      vllm_base_models:
+        - "./models/mistral/mistralai--Mistral-7B-Instruct-v0.1_a10g_tp1.yaml"
```
Comment (akshay-anyscale) [inline templates/endpoints/llm-serve.yaml:8]: let's leave the comment in for using l4 for GCP

File: templates/endpoints/README.md [hunk 2]
```diff
@@ -92,6 +92,7 @@ Look at the following guides for more advanced use-cases -
 * [Deploy models for embedding generation](EmbeddingModels.md)
 * [Learn how to bring your own models](CustomModels.md)
 * [Deploy multiple LoRA fine-tuned models](DeployLora.md)
+* [Deploy Function calling models](DeployFunctionCalling.md)
```
Comment (akshay-anyscale) [inline templates/endpoints/README.md:95]: dead link

Comment (gvspraveen) [reply templates/endpoints/README.md:95]: fixed

File: templates/endpoints/func_calling-serve.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+applications:
+  - name: llm-endpoint
+    route_prefix: /
+    import_path: aviary_private_endpoints.backend.server.run:router_application
+    args:
+      models: []
+      function_calling_models:
+        - "./models/mistral/mistralai--Mistral-7B-Instruct-v0.1_a10g_tp1.yaml"
```
Comment (akshay-anyscale) [inline templates/endpoints/func_calling-serve.yaml:8]: let's also add comment here for GCP

File: templates/endpoints/DeployFunctionCalling.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
+# Serving Function calling Models
+
+You can support function calling models by using following steps.
```
Comment (akshay-anyscale) [inline templates/endpoints/DeployFunctionCalling.md:3]: We should link to our blog and endpoints docs on function calling here to explain the mechanism

**Review by akshay-anyscale — APPROVED** [review]

**Pull request merged**